### PR TITLE
Quiz questions

### PR DESCRIPTION
**Pull request by omkarpatke**
# Added rules page category-wise questions and result page.

**[Preview](https://deploy-preview-3--faster-quizz.netlify.app/)**

**Files To Check:**
- QuizPage.js
- QuizPage.css
- RulesPage.js
- ResultPage.js

### CODE REVIEW

**Comment by omkarpatke** [comment]
> # Added rules page category-wise questions and result page.
> **[Preview](https://deploy-preview-3--faster-quizz.netlify.app/)**
> 
> **Files To Check:**
> 
> * QuizPage.js
> * QuizPage.css
> * RulesPage.js
> * ResultPage.js

File: src/Pages/QuizPage/QuizPage.css [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+.incrementBtn{
+    margin-left: 30rem;
+}
+
+.questionNumber{
+    display: flex;
+    gap: 30rem;
+    margin-top: 10px;
+    font-weight: 600;
+}
```
Comment (MonikaJhingan) [inline src/Pages/QuizPage/QuizPage.css:45]: consistent use of rem is appreciable.

File: src/Pages/QuizPage/QuizPage.css [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    background-color: aliceblue;
+    width: 100%;
+    padding: 10px 16rem;
+    font-size: 1.1rem;
+    cursor: pointer;
+}
+
+.option:hover{
+    box-shadow: 1px 2px 2px 2px rgba(0,0,0,0.1);
+}
```
Comment (MonikaJhingan) [inline src/Pages/QuizPage/QuizPage.css:34]: Use of var () for color is more advisable.

File: src/Pages/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            <button className="login-btn" type="submit" onClick={guestLoginHandler}> Login as Guest </button>
+            <h2>OR</h2>
+            <button className="signIn-with-google-btn"><i className="lni lni-google"></i> <span className='signInWithGoogleBtnText'>Sign in with google</span></button>
+
+            <p>I Don't Have An Account</p>
+            <Link to='/sign-up'>CREATE AN ACCOUNT</Link>
+        </div>
+    </div>
+    </>
+  )
```
Comment (MonikaJhingan) [inline src/Pages/Login/Login.js:60]: The quiz has been nicely set with guest credentials for login.

File: src/Pages/QuizPage/QuizPage.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    if(count === 4){
+      navigate('/quiz-result');
+      quizQuestionsDispatch({type : 'UPDATE_SCORE' , payload : quizQuestionsState.payload ,score: finalScore  } )
+      console.log(quizQuestionsState , finalScore)
+    }
+  }
+
+
+
```
Comment (MonikaJhingan) [inline src/Pages/QuizPage/QuizPage.js:30]: Pls avoid these extra gaps.

File: src/Pages/ResultPage/ResultPage.css [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    width: 45rem;
+}
+
+.questions{
+    display: flex;
+    flex-direction: column;
+}
+
+.currectAns{
+    background-color: rgb(56, 148, 56);
```
Comment (MonikaJhingan) [inline src/Pages/ResultPage/ResultPage.css:23]: Use of var() is more advisable for color.

File: src/Pages/ResultPage/ResultPage.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    const { quizQuestionsState } = useQuizQuestionsContext();
+    console.log(quizQuestionsState)
+  return (
+    <>
+    <div className='result-page-container'>
+    <div className="result-cart">
+    <h2>You Have Scored {quizQuestionsState.score} out of 10</h2>
+
+    <div className="questions">
+       {quizQuestionsState.payload.map(question => (
```
Comment (MonikaJhingan) [inline src/Pages/ResultPage/ResultPage.js:15]: Map() is well implemented.

File: src/Pages/SignUp/SignUp.css [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+.login-card{
+    display: flex;
+    flex-direction: column;
+    justify-content:center;
+    align-items: center;
+    padding: 1rem 3rem 2rem 3rem;
+    margin: 1rem 3rem;
+    border-radius: 10px;
+    box-shadow: 1px 1px 10px 10px rgba(0,0,0,0.1);
+    background-color: var(--white-color);
```
Comment (MonikaJhingan) [inline src/Pages/SignUp/SignUp.css:24]: good use of var() but pls be consistent for good code quality.

File: src/components/api-calls/apisCalls.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import axios from 'axios';
+
+export const getQuestions = () => {
+    const response = axios.get('https://opentdb.com/api.php?amount=10&category=21&difficulty=easy&type=multiple');
+    console.log(response)
```
Comment (MonikaJhingan) [inline src/components/api-calls/apisCalls.js:5]: api calling is nicely done.

**Review by MonikaJhingan — APPROVED** [review]
The code has well implemented logic and is written neatly keeping all smaller aspects in mind. It should definitely be merged.

**Pull request merged**